lib/core_extensions.rb

```diff
@@ -38,8 +38,12 @@ def initialize(*attribute)
     end
 
     def before_destroy(record)
-      klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
+      klasses.each do |klass, foreign_key|
+        record_class = record.class
+        association = record_class.reflect_on_association(klass)
+        foreign_key ||= "#{record_class.name}_id".to_sym
```

Comment:
`"#{record_class.name}_id".to_sym` will work most of the time.
For user/usergroup I used `owner_id` for host but that creates `Host.unscoped.where(:owner_id => 123)` which is not good enough since it's a polymorphic association.